metadata-ingestion/src/datahub/ingestion/source/snowflake/snowflake_v2.py

```diff
@@ -531,12 +531,20 @@ def get_workunits_internal(self) -> Iterable[MetadataWorkUnit]:
             for db in databases
             for schema in db.schemas
             for table_name in schema.tables
+            if self._is_dataset_pattern_allowed(
+                self.get_dataset_identifier(table_name, schema.name, db.name),
+                SnowflakeObjectDomain.TABLE,
+            )
```

Comment:
This should really be done when filtering `db.schemas.tables` right? Happy to leave this in though just in case